tools/osv-linter/internal/checks/checks.go

```diff
@@ -0,0 +1,140 @@
+// Package checks defines and implements all checks and collections of checks.
+//
+// To add additional checks:
+// 1. define a new instance of `Check`
+// 2. add it to the `checks` array
+// 3. add it to the relevent collections defined in `checkCollections`
+//
+// To add additional collections of checks:
+// 1. add to the `checkCollections` array.
+package checks
+
+import (
+	"fmt"
+
+	"github.com/tidwall/gjson"
+)
+
+// A CheckCode is a unique code for a check.
+type CheckCode string
```

Comment:
Is there any reason in particular why this is a new type, as opposed to just being a string?  It's also strange that the only way to access this from a `Checker` is via `CodeString` which returns a string. 

When would type checking on CheckCode be useful?

Reply:
I read somewhere in my Go learnings (I fail to recall where, nor find it again quickly) that it's more descriptive to have a custom string type than to have `map[string]whatever`, as it conveys more meaning as to _what_ the map's key is supposed to be.

Reply:
Hmm, in this case, is the map an implementation detail or part of the interface? It seems to fall to the former because of `CodeString` which is the way clients can retrieve this. 

If this is an implementation detail (i.e. map just happens to be what we use under the hood for some things), then exposing this to users is probably not ideal.

Reply:
I've reworked this part of the code since.